src/snowflake/snowpark/mock/connection.py

```diff
@@ -405,8 +410,29 @@ def execute(
         elif isinstance(res, list):
             rows = res
 
+        if to_pandas:
+            pandas_df = pd.DataFrame()
+            for col_name in res.columns:
+                pandas_df[unquote_if_quoted(col_name)] = res[col_name].tolist()
+            rows = _fix_pandas_df_integer(res)
+
+            # the following implementation is just to make DataFrame.to_pandas_batches API workable
+            # in snowflake, large data result are split into multiple data chunks
+            # and sent back to the client, thus it makes sense to have the generator
+            # however, local testing is designed for local testing
+            # we do not mock the splitting into data chunks behavior
+            if to_iter:
+                # we return a generator instead of iter(rows) because in the live connection
+                # rows is pd.DataFrame object, iter(rows) will return rows inside the pd.DataFrame which diverges
+                # from the live connection
+                def gen():
```

Comment:
I think we can replace this with `rows = [rows]`, then later code on line 433 can be reused.

Reply:
good idea! code updated